src/EditorFeatures/Core/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -293,7 +294,7 @@ internal sealed class CompletionSource : IAsyncExpandingCompletionSource
                     // and we'd like to have expanded items available when they are needed.
                     var expandedItemsTaskCancellationToken = _expandedItemsTaskCancellationSeries.CreateNext(cancellationToken);
                     var expandedItemsTask = Task.Run(() => GetCompletionContextWorkerAsync(session, document, trigger, triggerLocation,
-                                                                        options with { ExpandedCompletionBehavior = ExpandedCompletionMode.ExpandedItemsOnly },
+                                                                        options with { ExpandedCompletionBehavior = ExpandedCompletionMode.ExpandedItemsOnly, PerformSort = true },
```

Comment:
Consider adding a comment to explain why those needs to be sorted (i.e. we don't want to add sorting cost to item selection in ItemManager , which affect blocking and non-blocking typing scenarios, and we simply append unimported items to the list there with the assumption that they are already sorted)

Reply:
However. I wonder how much improvement this would bring, given that we are still sorting those unimported items as before, which constitute the majority of items

Reply:
Added a comment. 

I guess we'll see, but we've got to sort the expanded results at some point. BTW, the time frame in question is what was reported by a performance marker in a user trace indicating the time from completion request to it's initial presentation. I wouldn't think that would include time to compute the expanded results, but I'm not 100% sure.

Reply:
>  the time from completion request to it's initial presentation. I wouldn't think that would include time to compute the expanded results, but I'm not 100% sure.

By default it wouldn't. But there's option to wait for unimported items.
e.g.
https://github.com/dotnet/roslyn/blob/16af1971a71e6a249e15ca666c48fbf0c620a7de/src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs#L124